### PR TITLE
Add rejection reasons review to subject knowledge section

### PR DESCRIPTION
**Pull request by malcolmbaig**
## Context

As a.... candidate
In order to... improve my chance of success when applying again
I want to... review the sections of my application that providers said needed improvement.

## Changes proposed in this pull request
Building on https://github.com/DFE-Digital/apply-for-teacher-training/pull/4551, add to the 'subject_knowledge' section the same review workflow that we've implemented for 'becoming_a_teacher'. To summarize, this will:

- Show a review badge against this section if the candidate is applying
  again and they have subject knowledge rejection reasons from their
  previous application.
- Show past rejection reasons inline when they're reviewing
  subject knowledge.
- Update validations on application submit to refer to 'reviewing' the
  section rather than completing it.

This PR also includes several refactoring/housekeeping commits. 

## Guidance to review
- Per commit recommended.
- Manual testing will need an unsuccessful application with structured feedback in `quality_of_application_subject_knowledge_what_to_improve`. The review workflow can then be tested by applying again.

## Link to Trello card

https://trello.com/c/BkHTZAJE
## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/views/candidate_interface/subject_knowledge/show.html.erb [hunk 1]
```diff
@@ -5,11 +5,25 @@
   <%= t('page_titles.subject_knowledge') %>
 </h1>
 
+<% if @application_form.review_pending?(:subject_knowledge) %>
+  <div class='govuk-inset-text govuk-!-width-two-thirds govuk-!-margin-top-0'>
```
Comment (frankieroberto) [inline app/views/candidate_interface/subject_knowledge/show.html.erb:9]: I think we can use the `GovukComponent::InsetText` component here, now that it supports block content?

File: app/views/candidate_interface/subject_knowledge/show.html.erb [hunk 2]
```diff
@@ -5,11 +5,25 @@
...
+    <h2 class="govuk-heading-m">Feedback from previous applications</h2>
```
Comment (frankieroberto) [inline app/views/candidate_interface/subject_knowledge/show.html.erb:10]: I think it makes sense to pluralise this based on the number of bits of feedback shown?

```suggestion
    <h2 class="govuk-heading-m">Feedback from previous <%= pluralize(@application_form.subject_knowledge_rejection_reasons.size, "application") %></h2>
```

**Comment by malcolmbaig** [comment]
Going to close temporarily to fix some linting and to add newline handling to the inline rejection history we display when clicking through to the section.

File: app/views/candidate_interface/subject_knowledge/show.html.erb (hunk 1 above)
Comment (malcolmbaig) [reply app/views/candidate_interface/subject_knowledge/show.html.erb:9]: nice! 👍🏼

File: spec/system/candidate_interface/apply_again/candidate_reviews_rejection_reasons_from_previous_applications_spec.rb [hunk 3]
```diff
@@ -7,15 +7,17 @@
     given_i_am_signed_in_as_a_candidate
     and_i_have_an_unsuccessful_application_with_rejection_reasons
     when_i_apply_again
-    then_become_a_teacher_needs_review
-    and_i_can_review_become_a_teacher
+    then_subject_knowledge_needs_review
+    then_becoming_a_teacher_needs_review
```
Comment (stevehook) [inline spec/system/candidate_interface/apply_again/candidate_reviews_rejection_reasons_from_previous_applications_spec.rb:11]: Nit: `and_becoming_a_teacher_needs_review`?

**Review by stevehook — APPROVED** [review]
:+1: Tested review app and all works for me.

**Review by frankieroberto — APPROVED** [review]
Looks good, thanks.

**Pull request merged**